### PR TITLE
[DR-2180] Increase retry timeout in BigQuery tests

### PR DESCRIPTION
**Pull request by samanehsan**
All of the settings are the same as the default retrySettings except for the total timeout of 7 minutes, which is how long it can take for permissions to propagate.

### CODE REVIEW

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java [hunk 1]
```diff
@@ -35,11 +37,23 @@ public static BigQuery getClientForTestUser(
         "Fetching credentials and building BigQuery client object for test user: {}",
         testUser.name);
 
+    RetrySettings retrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRetryDelay(Duration.ofMillis(1000L))
+            .setMaxRetryDelay(Duration.ofMillis(32_000L))
```
Comment (nmalfroy) [inline datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java]: FWIW, I think you can do stuff like `Duration.ofMillis(TimeUnit.SECONDS.toMillis(32))` or `Duration.of(32, TimeUnit.SECONDS.toChronoUnit())` which is maybe a touch easier to read

**Review by nmalfroy — APPROVED** [review]
lgtm!  Just the one thing about how to get the time

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java (hunk 1 above)
Comment (pshapiro4broad) [reply datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java]: I think `Duration.ofSeconds()` works well here.
```suggestion
            .setInitialRetryDelay(Duration.ofSeconds(1))
            .setMaxRetryDelay(Duration.ofSeconds(32))
```

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java [hunk 2]
```diff
@@ -14,6 +15,7 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.threeten.bp.Duration;
```
Comment (pshapiro4broad) [inline datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java:18]: I think this is the wrong import, I think you want the java.time one.
```suggestion
import java.time.Duration;
```

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java (hunk 1 above)
Comment (samanehsan) [reply datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java]: Good point!

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java (hunk 2 above)
Comment (samanehsan) [reply datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java:18]: That's what I thought as well, but it's `org.threeten.bp.Duration`: https://github.com/googleapis/gax-java/blob/main/gax/src/main/java/com/google/api/gax/retrying/RetrySettings.java#L36

Comment (pshapiro4broad) [reply datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java:18]: Well, that's not confusing at all :( Apparently this is a library that provides JDK 8 time support for JDK 6 and 7. It's a shame it's not used in a way that's forward compatible; the code could accept either type and convert it.

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java [hunk 3]
```diff
@@ -35,11 +37,24 @@ public static BigQuery getClientForTestUser(
         "Fetching credentials and building BigQuery client object for test user: {}",
         testUser.name);
 
+    RetrySettings retrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRetryDelay(Duration.ofMillis(1000L))
+            .setInitialRetryDelay(Duration.ofSeconds(1))
```
Comment (fboulnois) [inline datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java]:
```suggestion
            .setInitialRetryDelay(Duration.ofSeconds(1))
```

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java [hunk 4]
```diff
@@ -35,11 +37,24 @@ public static BigQuery getClientForTestUser(
...
+            .setMaxRetryDelay(Duration.ofSeconds(32))
+            .setRetryDelayMultiplier(2.0)
+            .setTotalTimeout(Duration.ofMinutes(7))
+            .setInitialRpcTimeout(Duration.ofMillis(50_000L))
```
Comment (fboulnois) [inline datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java]:
```suggestion
            .setInitialRpcTimeout(Duration.ofSeconds(50))
```

File: datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java [hunk 5]
```diff
@@ -35,11 +37,24 @@ public static BigQuery getClientForTestUser(
...
+            .setRpcTimeoutMultiplier(1.0)
+            .setMaxRpcTimeout(Duration.ofMillis(50_000L))
```
Comment (fboulnois) [inline datarepo-clienttests/src/main/java/common/utils/BigQueryUtils.java]:
```suggestion
            .setMaxRpcTimeout(Duration.ofSeconds(50))
```

**Review by fboulnois — COMMENTED** [review]
Minor suggestions, they probably should be applied to `src/test/java/bio/terra/integration/BigQueryFixtures.java` too.

**Review by snf2ye — APPROVED** [review]

**Pull request merged**